src/firefly/js/visualize/ui/IrsaCatalogSearch.jsx

```diff
@@ -39,15 +39,16 @@ export function IrsaCatalogSearch() {
 
     return (
         <FieldGroup groupKey={irsaCatalogGroupKey} keepState={true} sx={{position:'relative', flexGrow:1}}>
-            <FormPanel sx={{position:'absolute', inset:'0 0 0 0', overflow:'hidden', maxWidth:'100em', margin:'0 auto'}}
-                groupKey={[irsaCatalogGroupKey, gkeySpacial]}
-                onSuccess={(request) => onSearchSubmit(request)}
-                onError={(request) => onSearchFail(request)}
-                params={{hideOnInvalid: false}}
+            <FormPanel sx={{position:'absolute', inset:0, overflow:'hidden', maxWidth:'100em', margin:'0 auto'}}
+                groupKey={() => ([irsaCatalogGroupKey, gkeySpacial])}
+                onSuccess={onSearchSubmit}
+                onError={onSearchFail}
                 help_id={'catalogs.irsacatalogs'}
-                cancelText=''
-                slotProps={{input: {overflow: 'auto'}}}
-            >
+                slotProps={{
+                    input: {overflow: 'auto'},
+                    cancelBtn: {component:null}
```

Comment:
In regards to `component=null`, I thought it was a clever way of reusing an existing reserved prop to turning on/off a slot.  We can create our own reserved prop to do that same, if that is important to you.  Just make sure it does not clash with other commonly used props, like `display`.

As for `useCancelButton: false`, I don't think it's a good idea.  This is how we end up with all top-level props to begin with.  If there is standard way of turning on/of a slot, then that's what we should use.

Reply:
Not everything that JoyUI does like this is via a slot. They do expose common props at the top level and use slots of less used settings.

Reply:
I will revert cancel button changes.  It will keep its original implementation without the ability to customize.  We will expose `onCancel` and `cancelText` at the top level as before.  When `cancelText` is nullish, then no cancel button will be rendered.  This is not how I would have done it.  But, it's inline with what you are asking for.

Reply:
Maybe changing the defaults for cancel btn (see my previous comment) can help with this too while keeping the slots/slotProps design (which should be kept IMO because that's how complete button is)